Wrap blocked domains and keywords in collapsible details element

obenland opened the pull request:
Before | After
--- | ---
<img width="1582" height="1034" alt="Screenshot 2025-12-04 at 3 53 12 PM" src="https://github.com/user-attachments/assets/61d2f86f-6cfa-46c1-a648-c41de8d6df45" /> | <img width="1582" height="1034" alt="Screenshot 2025-12-04 at 3 51 59 PM" src="https://github.com/user-attachments/assets/c30d847a-f4cf-4e3d-b633-210a4db147fd" />


## Proposed changes:

* Wrap blocked domains and keywords tables in collapsible `<details>` elements.
* Display count in summary (e.g., "82 blocked domains") so users know how many items exist.
* Keep add form always visible outside the collapsible section.
* Reduces visual clutter when many domains/keywords are blocked.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?

## Testing instructions:

* Go to Settings → ActivityPub → Moderation tab.
* Add several blocked domains or keywords.
* Verify the table is collapsed by default showing "X blocked domains/keywords".
* Click the summary to expand and see the full list.
* Verify the add form remains visible when collapsed.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Added - for new features
-   [x] Changed - for changes in existing functionality
-   [ ] Deprecated - for soon-to-be removed features
-   [ ] Removed - for now removed features
-   [ ] Fixed - for any bug fixes
-   [ ] Security - in case of vulnerabilities

#### Message

Wrap blocked domains and keywords tables in collapsible details element.

</details>

obenland committed:
Wrap blocked domains and keywords tables in collapsible details element

matticbot committed:
Add changelog

obenland committed:
Fix escaping for sprintf output

obenland commented on includes/wp-admin/class-settings-fields.php:
```diff
@@ -429,23 +429,35 @@ public static function render_moderation_section_description() {
-				<?php endforeach; ?>
-			</table>
+			<details style="margin: 10px 0;">
+				<summary style="cursor: pointer; padding: 8px 0; color: inherit;">
```
That's supported natively.

obenland commented on includes/wp-admin/class-settings-fields.php:
```diff
@@ -463,23 +475,35 @@ public static function render_site_blocked_domains_field() {
-				<?php endforeach; ?>
-			</table>
+			<details style="margin: 10px 0;">
+				<summary style="cursor: pointer; padding: 8px 0; color: inherit;">
```
That's supported natively.

pfefferle commented on includes/wp-admin/class-settings-fields.php:
```diff
@@ -429,23 +429,37 @@ public static function render_moderation_section_description() {
-					</tr>
-				<?php endforeach; ?>
-			</table>
+			<details style="margin: 10px 0;">
```
Please use the CSS file instead of inline CSS.

obenland replied:
Fixed in [4d67dae](https://github.com/Automattic/wordpress-activitypub/pull/2591/commits/4d67daef09b7d64252ff4053615c6d97f375c72c).

obenland committed:
Move inline CSS to stylesheet for site block details

Move the inline styles for the collapsible details/summary elements
to activitypub-admin.css instead of using inline styles in PHP.

pfefferle commented:
I generally like the idea, but it is still a bit glitchy.

* When I initially add new items, they will be added to the classic table (as it was before)
* After a reload the table is gone and the collapsible summary shows up.
* When I now add new items, neither the number of the summary raises, nor do I see if my addition was really added to the list.

obenland committed:
Fix collapsible moderation tables UX and use wp.i18n for translations

- Always render <details> wrapper structure (even when empty) with data-type attribute
- Add <tbody> wrapper for proper striped table styling
- Update JavaScript to work with collapsible structure and update summary counts dynamically
- Use wp.i18n._n() for proper pluralization instead of localized strings
- Add wp-i18n dependency and wp_set_script_translations()
- Extract duplicated validation/message code into helper functions
- Add JSDoc comments for better IDE support

obenland committed:
Merge branch 'trunk' into update/collapsible-moderation-tables

obenland committed:
Fix XSS vulnerability by using DOM-safe methods for user input

Use jQuery's element construction with attributes object and .text()
instead of string concatenation when inserting user-controlled values
into the DOM. This prevents malicious input from being executed.

obenland commented on assets/js/activitypub-moderation-admin.js:
```diff
@@ -27,29 +61,90 @@
 			}
-			table.append( '<tr><td>' + value + '</td><td style="width: 80px;"><button type="button" class="button button-small remove-site-block-btn" data-type="' + type + '" data-value="' + value + '">Remove</button></td></tr>' );
+
+			table.find( 'tbody' ).append( '<tr><td>' + value + '</td><td><button type="button" class="button button-small remove-site-block-btn" data-type="' + type + '" data-value="' + value + '">Remove</button></td></tr>' );
```
Fixed in [c0d132f](https://github.com/Automattic/wordpress-activitypub/pull/2591/commits/c0d132f60617eb9d2d3d88b80dcca97f578c7f25).

obenland commented:
@pfefferle This should behave less glitchy now.

obenland committed:
Merge branch 'trunk' into update/collapsible-moderation-tables
